src/OrchardCore.Modules/Orchard.Autoroute/Drivers/AutoroutePartDisplay.cs

```diff
@@ -61,14 +65,16 @@ public override IDisplayResult Edit(AutoroutePart autoroutePart)
         public override async Task<IDisplayResult> UpdateAsync(AutoroutePart model, IUpdateModel updater)
         {
             await updater.TryUpdateModelAsync(model, Prefix, t => t.Path);
-            
+
             var httpContext = _httpContextAccessor.HttpContext;
-            
             if (httpContext != null && await _authorizationService.AuthorizeAsync(httpContext.User, Permissions.SetHomepage))
             {
                 await updater.TryUpdateModelAsync(model, Prefix, t => t.SetHomepage);
             }
 
+            var validationErrorKeyPrefix = string.IsNullOrEmpty(Prefix) ? string.Empty : Prefix + ".";
```

Comment:
Sorry, my bad I have picked the wrong test case ... the validation message for numeric field isn't shown, but that's because there isn't any `<span asp-validation-for="Value"></span>` tag. For example the Link field works correctly.

In content field drivers the `Prefix` property contains the whole name of the HTML field, so the validation message is displayed correctly.

In content part drivers the `Prefix` property contains the name of the part e.g. `Autoroute` (IMO it is correct), but I need to display validation message for the `AutoRoute.Path` field. This behavior is the same even if I revert the change regarding `HtmlPrefix`.

I guess I can simply concatenate `Prefix + ".Path"` because inside part prefix shouldn't be `null`